src/messaging/ExchangeContext.cpp

```diff
@@ -157,8 +157,11 @@ CHIP_ERROR ExchangeContext::SendMessageImpl(Protocols::Id protocolId, uint8_t ms
         mReliableMessageContext.SetAckPending(false);
 
 #if !defined(NDEBUG)
-        ChipLogProgress(ExchangeManager, "Piggybacking Ack for MsgId:%08" PRIX32 " with msg",
-                        mReliableMessageContext.mPendingPeerAckId);
+        if (!sendFlags.Has(SendMessageFlags::kStandaloneAck))
```

Comment:
I guess this is at least not wrong, but I still don't understand why we can't just check the protocol id and message type instead of adding this extra flag what can now get out of sync with the actual message type being sent.

Reply:
Acked, we can just check the protocol id and message type